app/parse.py

```diff
@@ -2,11 +1,156 @@
+        course: Course
+) -> Course:
+    details = soup.select_one(".typography_landingH3__vTjok")["href"]
+    course_details_url = BASE_URL + details
+
+    num_topics = parse_num_topics(course_details_url)
+    num_modules = parse_num_modules(course_details_url)
+
+    course.num_topics = num_topics
+    course.num_modules = num_modules
+    return course
+
+
+def parse_num_topics(page_url: str) -> str:
+    page = requests.get(page_url).content
+    soup = BeautifulSoup(page, "html.parser")
+
+    num_topics = soup.select_one(
+        ".CourseProgram_purple__STsuM > .FactBlock_factNumber__d_8nn"
+    ).text
+
+    return num_topics
+
+
+def parse_num_modules(page_url: str) -> str:
+    driver = Driver.initialize()
+    driver.get(page_url)
+    find_button = driver.find_element(
+        By.CLASS_NAME,
+        "CourseProgram_modules__GA_PJ"
+    )
+    button_more = find_button.find_element(
+        By.CLASS_NAME,
+        "Button_neutral__ueX17"
+    )
+    to_scroll = driver.find_element(
+        By.CLASS_NAME,
+        "CourseProgram_modules__GA_PJ"
+    )
+    driver.execute_script("arguments[0].scrollIntoView();", to_scroll)
+
+    time.sleep(5)
+    button_more.click()
+
+    updated_page = driver.page_source
+    updated_soup = BeautifulSoup(updated_page, "html.parser")
+
+    num_modules = updated_soup.select(
+        ".CourseProgram_modules__GA_PJ > ul > li.color-dark-blue > "
+        ".CourseModulesList_moduleListItem__HKJqw > "
+        ".CourseModulesList_itemLeft__HP6K3 > "
+        "p.CourseModulesList_topicName__ZrDxT"
+    )
+
+    return str(len(num_modules))
+
+
+def get_all_courses() -> [Course]:
+    page = requests.get(BASE_URL + "/en/").content
+    soup = BeautifulSoup(page, "html.parser")
+
+    courses = soup.select(".ProfessionCard_cardWrapper__JQBNJ")
+    return [parse_single_course(course) for course in courses]
+
+
+def get_all_courses_with_extra_info() -> [Course]:
+    """this func was created to avoid github & selenium conflict"""
+
+    page = requests.get(BASE_URL + "/en/").content
+    soup = BeautifulSoup(page, "html.parser")
+
+    courses_info = soup.select(".ProfessionCard_cardWrapper__JQBNJ")
+    courses = []
+    for course_soup in courses_info:
+        course = parse_single_course(course_soup)
+        add_info = add_num_topics_and_num_modules(course_soup, course)
+        courses.append(add_info)
+
+    return courses
+
+
+def write_courses(output_csv_path: str, courses: list[Course]) -> None:
+    with open(output_csv_path, "w", newline="", encoding="utf8") as file:
+        writer = csv.writer(file)
+        writer.writerow([field.name for field in fields(Course)])
+        for course in courses:
+            writer.writerow(
+                [
+                    course.name,
+                    course.short_description,
+                    course.duration,
+                    course.num_topics,
+                    course.num_modules
+                ]
+            )
+
```

Comment:
This method writes a list of courses to a CSV file. However, it doesn't handle any potential CSV errors. For example, this method will throw an exception if a course object contains a value that can't be written to a CSV file. It's a good practice to always handle CSV errors when working with the `csv` module. You can do this by wrapping this code in a try/except block, and then handling any `csv.Error` exceptions that might be thrown.